### DISCUSSION TITLE
Setting gas as percent with a button or slider

### DISCUSSION THREAD

**Discussion by Dgocof** [artisan-roaster-scope/artisan#1150] [General]
I'm using a gas roaster with a modulation gas valve and a Fuji controller. I would like to create a button or slider to set the gas to a certain percent of maximum. So at the beginning of the roast I would like to have the burner on 25% for example. Is this possible? Could I get some advice?

**Comment by bigl00z3** [comment]
Would the following fit your need?

- Change the Fuji to a fixed value (not controlled by PID algorithm or SV value) . You can most likely do it by switching it to the manual mode. check for the register in the Fuji Modbus manual  (eg. for the PXF manual check [Here](https://felib.fujielectric.co.jp/download/ContentDownload.htm?dataid=16281277&site=global&lang=en) )
- Send the required MV value (25% in your case) using a button with a Modbus command
(register is 0079 in HEX so [121 in Decimal ](https://www.rapidtables.com/convert/number/hex-to-decimal.html?x=00F0) ) and the value would be value to 25% you  probably need to enter 2500 (but double check if there is a multiplication factor)

<img width="951" alt="image" src="https://user-images.githubusercontent.com/8886922/232585906-91b3d53d-7213-4686-9710-6393199dc7e9.png">

In this case, the command would probably be `write(1,121,2500)` (if your slave number for the Fuji is 1, otherwise change accordingly)

- When required reactivate the PID regulation, send a command to switch the Fuji to manual


Note: I am not in front of the roaster to test this, so you may have to test if it works as planned

Nico

**Comment by Dgocof** [comment]
Thanks Nico for your help. I will try this next week when I return.

**Comment by Dgocof** [comment]
Hi Nico,
This worked great. I had to initiate standby mode first. Then implement the
gas percent.

Carl Rand
Durango Coffee Company
Durangocoffee.com
970-385-0276 ext 6
